### PR TITLE
[build] use pip install of later setup.py to install test dependencies

### PR DESCRIPTION
**Pull request by liushilongbuaa**
#### Why I did it
Using 'python setup.py test' to install dependencies doesn't meet secure requirement.
#### How I did it
Use 'pip install' instead of setup.py to install dependencies.
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

### CODE REVIEW

File: slave.mk [hunk 1]
```diff
@@ -863,9 +863,20 @@ $(addprefix $(PYTHON_WHEELS_PATH)/, $(SONIC_PYTHON_WHEELS)) : $(PYTHON_WHEELS_PA
...
+		if [ ! "$($*_TEST)" = "n" ]; then \
+			python$($*_PYTHON_VERSION) setup.py test --dry-run; \
+			echo HHHHH;ls .eggs || true; \
+			dependencies=$$(ls .eggs/ | grep '.egg$$' | awk -F- '{print$$1}'); \
+			[[ "$$dependencies" == "" ]] || pip$($*_PYTHON_VERSION) install $$dependencies; \
+			sudo rm -rf .eggs/; \
+			python$($*_PYTHON_VERSION) setup.py test $(LOG); \
+			echo hhhhh;ls .eggs || true; \
+		fi
+		echo ==================== end test 1>&2
```
Comment (xumia) [inline slave.mk:877]: We need to run line 879 before testing, right?

**Review by qiluo-msft — CHANGES_REQUESTED** [review]
Offline discussed. This issue is that dev is `python3 setup.py test` in sub repo's folder directly.

**Comment by liushilongbuaa** [comment]
> Offline discussed. This issue is that dev is `python3 setup.py test` in sub repo's folder directly.

We can discuss this issue next week. But this PR still can be merged as a fix for deepdiff's root cause.

File: slave.mk [hunk 2]
```diff
@@ -865,7 +865,7 @@ $(addprefix $(PYTHON_WHEELS_PATH)/, $(SONIC_PYTHON_WHEELS)) : $(PYTHON_WHEELS_PA
 ifneq ($(CROSS_BUILD_ENVIRON),y)
 		# Use pip instead of later setup.py to install dependencies into user home, but uninstall self
 		pip$($*_PYTHON_VERSION) install . && pip$($*_PYTHON_VERSION) uninstall --yes `python$($*_PYTHON_VERSION) setup.py --name`
-		if [ ! "$($*_TEST)" = "n" ]; then python$($*_PYTHON_VERSION) setup.py test $(LOG); fi
+		if [ ! "$($*_TEST)" = "n" ]; then python$($*_PYTHON_VERSION) $(BUILD_WORKDIR)/scripts/get_py_depend.py | xargs -i pip$($*_PYTHON_VERSION) install {}; python$($*_PYTHON_VERSION) setup.py test; fi $(LOG)
```
Comment (xumia) [inline slave.mk:868]: How about moving the logic inside of the python program get_py_depend.py (give a better name).

**Pull request closed without merging**